rest-api-spec/src/main/resources/rest-api-spec/test/cluster.voting_config_exclusions/10_basic.yml

```diff
@@ -0,0 +1,137 @@
+teardown:
+  - do:
+      cluster.delete_voting_config_exclusions: {}
+
+---
+"Get cluster state without voting config exclusions":
+  - do:
+      cluster.state: {}
+
+  - length: { metadata.cluster_coordination.voting_config_exclusions: 0 }
+
+---
+"Add voting config exclusion by unknonw node Id":
+  - do:
+      cluster.post_voting_config_exclusions:
+        node_ids: nodeId
+
+  - do:
+      cluster.state: {}
+
+  - length: { metadata.cluster_coordination.voting_config_exclusions: 1 }
+  - match: { metadata.cluster_coordination.voting_config_exclusions.0.node_id: "nodeId" }
+  - match: { metadata.cluster_coordination.voting_config_exclusions.0.node_name: "_absent_" }
+
+---
+"Add voting config exclusion by unknonw node Ids":
+  - skip:
+        reason: "contains is a newly added assertion"
+        features: contains
+
+  - do:
+      cluster.post_voting_config_exclusions:
+        node_ids: nodeId1,nodeId2
+
+  - do:
+      cluster.state: {}
+
+  - length: { metadata.cluster_coordination.voting_config_exclusions: 2 }
+  - contains : { metadata.cluster_coordination.voting_config_exclusions: {node_id: "nodeId1", node_name: "_absent_"} }
+  - contains : { metadata.cluster_coordination.voting_config_exclusions: {node_id: "nodeId2", node_name: "_absent_"} }
+
+---
+"Add voting config exclusion by unknonw node name":
+  - do:
+      cluster.post_voting_config_exclusions:
+        node_names: nodeName
+
+  - do:
+      cluster.state: {}
+
+  - length: { metadata.cluster_coordination.voting_config_exclusions: 1 }
+  - match: { metadata.cluster_coordination.voting_config_exclusions.0.node_id: "_absent_" }
+  - match: { metadata.cluster_coordination.voting_config_exclusions.0.node_name: "nodeName" }
+
+---
+"Add voting config exclusion by unknonw node names":
+  - skip:
+      reason: "contains is a newly added assertion"
+      features: contains
+
+  - do:
+      cluster.post_voting_config_exclusions:
+        node_names: nodeName1,nodeName2
+
+  - do:
+      cluster.state: {}
+
+  - length: { metadata.cluster_coordination.voting_config_exclusions: 2 }
+  - contains : { metadata.cluster_coordination.voting_config_exclusions: {node_id: "_absent_", node_name: "nodeName1"} }
+  - contains : { metadata.cluster_coordination.voting_config_exclusions: {node_id: "_absent_", node_name: "nodeName2"} }
+
+#---
+# THIS TEST CASE WILL CAUSE TIMEOUT FOR CLEAR VOTING CONFIG OPERATION, HENCE BREAKING ALL TESTS
```

Comment:
For this test and the next, as there's only 1 master node in the cluster, I tried to post a voting config exclusion with that master node id / name. Doing so caused timeout exceptions for these tests themselves as well as others when executing clearing voting config exclusion in teardown. Not sure if there's a good workaround for this issue? I searched around to see if I can set up more nodes in the cluster for testing, but haven't found anything yet.

Reply:
Yes, waiting and then timing out is what I'd expect here. For the clear operation you can avoid the wait with `wait_for_removal: true` but there's no way to avoid the wait during the add operation. I think it'd be best to use a bogus node name or ID instead.

Reply:
I see. For the clear operation I do need it to complete successfully to reset cluster voting config exclusion state for the next test, so I guess there's no obvious work around then. 

Since I already have other tests to use bogus node name or id, I'll remove these two tests   then.